src/usernotifications.cs

```diff
@@ -0,0 +1,653 @@
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.TvOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotification : NSCopying, NSSecureCoding {
+
+		[Export ("date", ArgumentSemantic.Copy)]
+		NSDate Date { get; }
+
+		 [Export ("request", ArgumentSemantic.Copy)]
+		 UNNotificationRequest Request { get; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotificationAction : NSCopying, NSSecureCoding {
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("title")]
+		string Title { get; }
+
+		[Export ("options", ArgumentSemantic.Assign)]
+		UNNotificationActionOptions Options { get; }
+
+		[Static]
+		[Export ("actionWithIdentifier:title:options:")]
+		UNNotificationAction FromIdentifier (string identifier, string title, UNNotificationActionOptions options);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (UNNotificationAction))]
+	interface UNTextInputNotificationAction {
+
+		[Static]
+		[Export ("actionWithIdentifier:title:options:textInputButtonTitle:textInputPlaceholder:")]
+		UNTextInputNotificationAction FromIdentifier (string identifier, string title, UNNotificationActionOptions options, string textInputButtonTitle, string textInputPlaceholder);
+
+		[Export ("textInputButtonTitle")]
+		string TextInputButtonTitle { get; }
+
+		[Export ("textInputPlaceholder")]
+		string TextInputPlaceholder { get; }
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UNNotificationAttachment : NSCopying, NSSecureCoding {
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("URL", ArgumentSemantic.Copy)]
+		NSUrl Url { get; }
+
+		[Export ("type")]
+		string Type { get; }
+
+		[Static]
+		[Export ("attachmentWithIdentifier:URL:options:error:")]
+		[return: NullAllowed]
+		UNNotificationAttachment FromIdentifier (string identifier, NSUrl url, [NullAllowed] NSDictionary options, [NullAllowed] out NSError error);
+	}
+
+	[Introduced (PlatformName.iOS, 10, 0)]
+	[Introduced (PlatformName.WatchOS, 3, 0)]
+	[Unavailable (PlatformName.TvOS)]
+	[Static]
+	interface UNNotificationAttachmentOptionsKeys {
+
+		[Field ("UNNotificationAttachmentOptionsTypeHintKey")]
+		NSString TypeHint { get; }
+
+		[Field ("UNNotificationAttachmentOptionsThumbnailHiddenKey")]
+		NSString ThumbnailHidden { get; }
+
+		[Field ("UNNotificationAttachmentOptionsThumbnailClippingRectKey")]
+		NSString ThumbnailClippingRect { get; }
+
+		[Field ("UNNotificationAttachmentOptionsThumbnailTimeKey")]
+		NSString ThumbnailTime { get; }
+	}
+
+#if XAMCORE_2_0 && !TVOS
```

Comment:
why `!TVOS` when we already have `[Unavailable (PlatformName.TvOS)]` ? generator bug ?